### PR TITLE
Mo's algorithm Problem

### PR DESCRIPTION
**Pull request by brebenelmihnea**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [ ] I have tested my code.
- [ ] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [ ] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: content/5_Plat/Sqrt.mdx [hunk 1]
```diff
@@ -359,6 +355,79 @@ public class DRSQ {
...
+vector<query> queries;
+
+int main() {
+	int n;
+	cin >> n;
+	for (int i = 1; i <= n; i++) { cin >> v[i]; }
+
+	int q;
+	cin >> q;
+	for (int i = 1; i <= q; i++) {
```
Comment (SansPapyrus683) [inline content/5_Plat/Sqrt.mdx:384]: is 1-indexing needed

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/5_Plat/Sqrt.mdx [hunk 2]
```diff
@@ -359,6 +355,75 @@ public class DRSQ {
...
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+const int NMAX = 3e5 + 1;
+const int VALMAX = 1e6 + 1;
+
+struct query {
+	int l, r, idx;
+};
```
[the remark this answers is not in the document]
Comment (brebenelmihnea) [reply content/5_Plat/Sqrt.mdx:370]: Ok

File: content/5_Plat/Sqrt.mdx [hunk 3]
```diff
@@ -359,6 +355,75 @@ public class DRSQ {
...
+	cin >> n;
+	for (int i = 1; i <= n; i++) { cin >> v[i]; }
+	cin >> q;
+	for (int i = 1; i <= q; i++) {
+		int x, y;
+		cin >> x >> y;
+		queries.push_back({x, y, i});
+	}
+
+	int BLOCK_SIZE = (int)sqrt(n);
```
[the remark this answers is not in the document]
Comment (brebenelmihnea) [reply content/5_Plat/Sqrt.mdx:390]: Ok

File: content/5_Plat/Sqrt.mdx [hunk 4]
```diff
@@ -359,6 +355,79 @@ public class DRSQ {
...
+const int VALMAX = 1e6 + 1;
+
+struct query {
+	int l, r, idx;
+};
+
+int different_values;
+int v[NMAX];
+int cnt[VALMAX];
+vector<query> queries;
```
[the remark this answers is not in the document]
Comment (brebenelmihnea) [reply content/5_Plat/Sqrt.mdx:375]: Ok

File: content/5_Plat/Sqrt.mdx (hunk 1 above)
Comment (brebenelmihnea) [reply content/5_Plat/Sqrt.mdx]: All right

**Comment by SansPapyrus683** [comment]
bro why'd u revert all my changes :skull:

**Comment by brebenelmihnea** [comment]
That was the fastest response :))
I will change it

**Review by SansPapyrus683 — APPROVED** [review]

**Review by ryanchou-dev — APPROVED** [review]
nicee

**Pull request merged**